comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +24,204 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+            long directorySize = 0;
+            Stack<string> directoriesLeft = new Stack<string>();
+
+            if (IsDirectory(directory))
+            {
+                directoriesLeft.Push(directory);
+
+                while (directoriesLeft.Count > 0)
+                {
+                    string currentDirectory = directoriesLeft.Pop();
+                    string[] newDirectories = Directory.GetDirectories(currentDirectory);
+                    string[] newfiles = Directory.GetFiles(currentDirectory);
+
+                    for (int i = 0; i < newDirectories.Length; i++)
+                    {
+                        directoriesLeft.Push(newDirectories[i]);
+                    }
```

Comment:
This code is repeated almost verbatim in all of your functions -- could you have refactored it into a helper function just once? E.g. a `Traverse` method that gets a list of files/directories which your other functions then iterate over